pkg/vcs/vcs.go

```diff
@@ -58,3 +87,86 @@ func VCS(path string) (result v1.VCS) {
 
 	return
 }
+
+func ImageInfoFromApp(ctx context.Context, app *apiv1.App) (string, string, error) {
+	vcs := app.Status.Staged.AppImage.VCS
+	if len(vcs.Remotes) == 0 {
+		return "", "", fmt.Errorf("clone can only be done on an app built from a git repository")
+	}
+	if vcs.Acornfile == "" {
+		return "", "", fmt.Errorf("app has no acornfile information in vcs")
+	}
+
+	// Create auth object to use when fetching and cloning git repos
+	auth, err := ssh.NewSSHAgentAuth("git")
+	if err != nil {
+		return "", "", err
+	}
+
+	for _, remote := range vcs.Remotes {
+		// Since we use ssh auth to clone the repo we need a git url but will sometimes get http urls
+		var gitUrl string
+		httpUrl, err := url.Parse(remote)
+		if err == nil {
+			gitUrl = fmt.Sprintf("git@%s:%s", httpUrl.Host, httpUrl.Path[1:])
+		} else {
+			gitUrl = remote
+		}
+
+		// Determine the repository name from the repo url
+		idx := strings.LastIndex(remote, "/")
+		if idx < 0 || idx >= len(remote) {
+			fmt.Printf("failed to determine repository name %q\n", remote)
+			continue
+		}
+		workdir := filepath.Clean(strings.TrimSuffix(remote[idx+1:], ".git"))
+
+		// Clone git repo and checkout revision
+		fmt.Printf("# Cloning repository %q into directory %q\n", gitUrl, workdir)
+		repo, err := git.PlainCloneContext(ctx, workdir, false, &git.CloneOptions{
+			URL:      gitUrl,
+			Auth:     auth,
+			Progress: os.Stderr,
+		})
+		if err != nil {
+			fmt.Printf("failed to clone repository %q: %s\n", gitUrl, err.Error())
+			continue
+		}
+		w, err := repo.Worktree()
+		if err != nil {
+			fmt.Printf("failed to get worktree from repository %q: %s\n", workdir, err.Error())
+			continue
+		}
+		err = w.Checkout(&git.CheckoutOptions{
+			Hash: plumbing.NewHash(vcs.Revision),
+		})
+		if err != nil {
+			fmt.Printf("failed to checkout revision %q for repository %q: %s\n", vcs.Revision, workdir, err.Error())
+			continue
+		}
+
+		// Create the Acornfile in the repository
+		acornfile := filepath.Join(workdir, vcs.Acornfile)
+		err = os.WriteFile(acornfile, []byte(app.Status.Staged.AppImage.Acornfile), 0666)
```

Comment:
The `Staged` field of the status is for image information that may not be set yet. I think this should be `app.Status.AppImage.Acornfile`.

Now that I think about it: is writing the Acornfile necessary? Are we not assuming that the Acornfile is in the repo?

Reply:
The idea was to cover the edge case that the Acornfile is either not in the repository or it exists in the repository in a different state than what was built into the running application